scico/examples.py

```diff
@@ -317,3 +317,52 @@ def spnoise(img: Array, nfrac: float, nmin: float = 0.0, nmax: float = 1.0) -> A
         imgn = imgn.at[spm < nfrac - 1.0].set(nmin)
         imgn = imgn.at[spm > 1.0 - nfrac].set(nmax)
     return imgn
+
+
+def create_3D_phantom_foam(
+    im_shape: Shape,
+    N_sphere: int,
+    r_mean: float = 0.1,
+    r_std: float = 0.001,
+    pad: float = 0.01,
+    is_random: bool = False,
+):
+    """Construct a 3D phantom with random radii and centers
+
+    Args:
+        im_shape: Shape of input image.
+        N_sphere: number of spheres added
+        r_mean: Mean radius of sphere (normalized to 1 along each axis).
+                Default 0.1.
+        r_std: Standard deviation radius of sphere (normalized to 1
+                along each axis). Default 0.001.
+        pad: Padding length (normalized to 1 along each axis). Default 0.01.
+        is_random: If False, random seed is set to 1 in order to make the
+                    process deterministic. Default: False.
+
+    Returns:
+        3D phantom of shape im_shape
+    """
+    c_lo = 0.0
+    c_hi = 1.0
+
+    if not is_random:
+        np.random.seed(1)
+
+    coord_list = [snp.linspace(0, 1, N) for N in im_shape]
+    x = snp.stack(snp.meshgrid(*coord_list, indexing="ij"), axis=-1)
+
+    print(x.shape)
+    print(im_shape)
+
+    centers = np.random.uniform(low=r_mean + pad, high=1 - r_mean - pad, size=(N_sphere, 3))
+    radii = r_std * np.random.randn(N_sphere) + r_mean
+
+    im = snp.zeros(im_shape) + c_lo
+    for c, r in zip(centers, radii):
+        dist = snp.sum((x - c) ** 2, axis=-1)
+        if snp.mean(im[dist < r ** 2] - c_lo) < 0.01 * c_hi:
+            im = im.at[dist < r ** 2].set(c_hi)
+            # im[dist < r**2] = c_hi
```

Comment:
Commented out code, or an explanation of the preceding line? If the latter, better to put it above that line?

Reply:
Put the comment above the line. I can remove it also if you think so.

Since the jax syntax `.at` and `.set` is perhaps an uncommon syntax, I added a comment stating how someone would write it in numpy so that it is clear what the line is doing.

Reply:
I think the comment is useful for those not too familiar with jax, but a few words of text might help clarify that it's intended as an explanation, and not commented-out code.